mirdata/guitarset.py

```diff
@@ -0,0 +1,391 @@
+        key_mode (KeyData)
+        pitch_contours (list): [(F0Data)s]
+            a list that contains 6 `F0Data`s. 
+            From Low E string to high e string.
+        notes (list): [(NoteData)s]
+            a list that contains 6 `NoteData`s. 
+            From Low E string to high e string.
+        audio (tuple): (np.ndarray, sr)
+    """
+
+    def __init__(self, track_id, data_home=None):
+        if track_id not in INDEX:
+            raise ValueError('{} is not a valid track ID in GuitarSet'.format(track_id))
+
+        self.track_id = track_id
+
+        if data_home is None:
+            data_home = utils.get_default_dataset_path(DATASET_DIR)
+
+        self._data_home = data_home
+        self._track_paths = INDEX[track_id]
+
+        self.audio_hex_cln_path = os.path.join(
+            self._data_home, self._track_paths['audio_hex_cln'][0]
+        )
+        self.audio_hex_path = os.path.join(
+            self._data_home, self._track_paths['audio_hex'][0]
+        )
+        self.audio_mic_path = os.path.join(
+            self._data_home, self._track_paths['audio_mic'][0]
+        )
+        self.audio_mix_path = os.path.join(
+            self._data_home, self._track_paths['audio_mix'][0]
+        )
+        self.jams_path = os.path.join(self._data_home, self._track_paths['jams'][0])
+
+        title_list = track_id.split('_')  # [PID, S-T-K, mode, rec_mode]
+        style, tempo, _ = title_list[1].split('-')  # [style, tempo, key]
+        self.player_id = title_list[0]
+        self.mode = title_list[2]
+        self.tempo = float(tempo)
+        self.style = _STYLE_DICT[style[:-1]]
+
+    # this lets users run `print(Track)` and get actual information
+    def __repr__(self):
+        repr_string = "GuitarSet Track(track_id={}, jams_path={})"
+        return repr_string.format(self.track_id, self.jams_path)
+
+    # `annotation` will behave like an attribute, but it will only be loaded
+    # and saved when someone accesses it. Useful when loading slightly
+    # bigger files or for bigger datasets. By default, we make any time
+    # series data loaded from a file a cached property
+    @utils.cached_property
+    def beats(self):
+        return _load_beats(self.jams_path)
+
+    @utils.cached_property
+    def leadsheet_chords(self):
+        if self.mode == 'solo':
+            logging.info(
+                'Chord annotations for solo excerpts are the same with the comp excerpt.'
+            )
+        return _load_chords(self.jams_path, leadsheet_version=True)
+
+    @utils.cached_property
+    def infered_chords(self):
+        if self.mode == 'solo':
+            logging.info(
+                'Chord annotations for solo excerpts are the same with the comp excerpt.'
+            )
+        return _load_chords(self.jams_path, leadsheet_version=False)
+
+    @utils.cached_property
+    def key_mode(self):
+        return _load_key_mode(self.jams_path)
+
+    @utils.cached_property
+    def pitch_contours(self):
```

Comment:
What about returning a dictionary keyed by string number/string name instead of a list? e.g.

```python
{
    'E': F0Data(...),
    'A': F0Data(...),
    ...
    'e': F0Data(...)
}
```

Reply:
I dig it.